From a267171dafde10f51058fdcafc4ce71a33213060 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Wed, 24 Jan 2024 15:50:43 +0100
Subject: [PATCH 464/841] feat(st-crypto): check if SAES is already initialized
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This patch checks if SAES has already been initialized to avoid
initializing it twice.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 6a237b36612c316ea7b6c48cbf5eb34cdcca775a feat(st-crypto): check if SAES is already initialized]
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Change-Id: Icd635735ff85eadaf386c9205b8215d515615680
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/354039
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
---
 drivers/st/crypto/stm32_saes.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/st/crypto/stm32_saes.c b/drivers/st/crypto/stm32_saes.c
index f4da571fb..fb3b13c6d 100644
--- a/drivers/st/crypto/stm32_saes.c
+++ b/drivers/st/crypto/stm32_saes.c
@@ -380,6 +380,11 @@ int stm32_saes_driver_init(void)
 {
 	int err;
 
+	if(saes_pdata.base != 0U) {
+		/* Driver is already initialized */
+		return 0;
+	}
+
 	err = stm32_saes_parse_fdt(&saes_pdata);
 	if (err != 0) {
 		return err;
-- 
2.47.2

